### PR TITLE
Nitpicking in the API pagination stuff

### PR DESCRIPTION
**Pull request by peterbourgon**
Refactor the `goto` away because I'm OCD. Avoid the dep because (a) just because, but also (b) I was able to get it to panic pretty easily with some examples directly from the RFC.

This is a draft because it needs more tests, but feel free to take a look.

### CODE REVIEW

**Review by phamann — APPROVED** [review]
> Avoid the dep because (a) just because, but also (b) I was able to get it to panic pretty easily with some examples directly from the RFC.

Hahaha, my spidey sense knew you wouldn't have liked the addition of the dep 😜 

---

👍🏻 LGTM so far.

**Comment by peterbourgon** [comment]
@phamann Ball in your court.

File: pkg/api/common.go [hunk 1]
```diff
@@ -35,3 +35,62 @@ func (e *Error) Error() string {
...
+			if !strings.EqualFold(val, "next") {
+				continue
+			}
+
+			// The URI we captured previously is the next link.
+			isRelNext = true
+			break
+		}
+
+		if isRelNext && linkURI != "" { // TODO(pb): parse URI for validity?
```
Comment (phamann) [inline pkg/api/common.go:90]: Per your todo comment, do we want to check URL validity here? In the context of this application I'm ok without, as we can "trust" the Fastly api to be returning valid link urls, but there is also no harm in adding it. Up to you.

**Review by phamann — APPROVED** [review]
LGTM other than comment 👍🏻

File: pkg/api/common.go (hunk 1 above)
Comment (peterbourgon) [reply pkg/api/common.go]: Oof, this is tricky. [Per the RFC](https://datatracker.ietf.org/doc/html/rfc5988#section-5.1)

> If the URI-Reference is relative, parsers MUST resolve it as per [RFC3986], Section 5. 

I don't think any of the extant packages do this. I'll write it up in a second.

**Comment by phamann** [comment]
Wow some beautiful refactoring, thank you. This looks good to merge now if you're ok with it @peterbourgon?

**Comment by peterbourgon** [comment]
@phamann Flattery will get you everywhere. Go for it.

**Pull request merged**